app/src/main/java/de/danoeh/antennapod/activity/MediaplayerActivity.java

```diff
@@ -823,6 +830,10 @@ void setupGUI() {
 
         SharedPreferences prefs = getSharedPreferences(PREFS, MODE_PRIVATE);
         showTimeLeft = prefs.getBoolean(PREF_SHOW_TIME_LEFT, false);
+        TimeSpeedConverter timeSpeedConverter = new TimeSpeedConverter(
+                UserPreferences.isTimeRespectSpeed(),
+                Float.parseFloat(UserPreferences.getPlaybackSpeed()));
```

Comment:
I am still not totally happy with that converter. It looks kind of "bloated". @mfietz Do you think removing the parameters and making `TimeSpeedConverter` read preferences on its own could be better?

Reply:
I'm used to avoid "singleton"/"static class" usage in "lower level" object, this way if we want to test the `TimeSpeedConverter` we don't have to mock the preference
And it is obvious to the reader the time conversion is based on preference.

However, I see your point, removing those parameters by making `TimeSpeedConverter` reading the preferences will make the code less bloated.

Let me know what you guys prefer, and I'll change it if needed.